flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolService.java

```diff
@@ -134,7 +139,25 @@ public final void start(
      *
      * @param componentMainThreadExecutor componentMainThreadExecutor used by this slot pool service
      */
-    protected void onStart(ComponentMainThreadExecutor componentMainThreadExecutor) {}
+    protected void onStart(ComponentMainThreadExecutor componentMainThreadExecutor) {
+        this.componentMainThreadExecutor = componentMainThreadExecutor;
+
+        componentMainThreadExecutor.schedule(
+                this::checkIdleSlotTimeout,
+                idleSlotTimeout.toMilliseconds(),
+                TimeUnit.MILLISECONDS);
```

Comment:
IIUC, you want to move the idle slot check logic from `AdaptiveScheduler` to `DeclarativeSlotPoolService`. But you didn't remove the code there, this is what confuses me. And It seems that this change is not completely aligned with FLINK-32399, is this safe?

Reply:
Sorry, my mistake. It should be [FLINK-31399]